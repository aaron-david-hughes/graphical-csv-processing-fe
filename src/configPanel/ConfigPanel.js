import React from 'react';
import Collapsible from "../generalPurposeComponents/collapsible/Collapsible";
import GraphController from "../graphController/GraphController";
import ProcessForm from "../processForm/ProcessForm";

class ConfigPanel extends React.Component {
    render() {
        return <div
            id='control'
            style={{
                width: '25%',
                display: 'block',
                margin: '1rem'
            }}
        >
            <Collapsible
                title='Edit Graph'
                id='nodeAndEdgeEditingCollapsible'
                style={{
                    marginBottom: '1rem'
                }}
            >
                <GraphController
                    addNode={this.props.addNode}
                    editNode={this.props.editNode}
                    deleteNode={this.props.deleteNode}
                    addEdge={this.props.addEdge}
                    editEdge={this.props.editEdge}
                    deleteEdge={this.props.deleteEdge}
                    addFile={this.props.addFile}
                    graphData={this.props.graphData}
                    setStep={this.props.setStep}
                    step={this.props.step}
                    editingId={this.props.editingId}
                />
            </Collapsible>

            <Collapsible
                title='Process'
                id='processCollapsible'
                style={{
                    marginBottom: '1rem'
                }}
            >
                {/*will have a file name option which will not be required*/}
                <ProcessForm
                    onSubmitForm={this.props.onSubmitForm}
                />
            </Collapsible>
        </div>;
    }
}

export default ConfigPanel;