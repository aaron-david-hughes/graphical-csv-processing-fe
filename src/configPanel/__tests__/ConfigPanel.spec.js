import React from 'react';
import {shallow} from 'enzyme';
import ConfigPanel from "../ConfigPanel";

describe('ConfigPanel tests', () => {
    let props = {
        config: jest.fn(),
        addNode: jest.fn(),
        addFile: jest.fn(),
        graphData: jest.fn(),
        addBanner: jest.fn(),
        onSubmitForm: jest.fn(),
        isGraphValid: true,
        loading: jest.fn(),
        isLoading: true
    };

    let render;

    beforeEach(() => {
        render = shallow(<ConfigPanel {...props} />);
    });

    afterEach(() => {
        render.unmount();
    });

    describe('ConfigPanel render tests', () => {
        it('shouldMatchSnapshot', () => {
            expect(render).toMatchSnapshot();
        });
    });
});