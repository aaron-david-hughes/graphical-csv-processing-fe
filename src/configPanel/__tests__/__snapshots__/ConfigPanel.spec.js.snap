// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigPanel tests ConfigPanel render tests shouldMatchSnapshot 1`] = `
<div
  id="control"
  style={
    Object {
      "display": "block",
      "margin": "1rem",
      "width": "25%",
    }
  }
>
  <Collapsible
    id="nodeAndEdgeEditingCollapsible"
    style={
      Object {
        "marginBottom": "1rem",
      }
    }
    title="Nodes"
  >
    <AddNodeController
      addBanner={[MockFunction]}
      addFile={[MockFunction]}
      addNode={[MockFunction]}
      config={[MockFunction]}
      graphData={[MockFunction]}
    />
  </Collapsible>
  <Collapsible
    id="processCollapsible"
    style={
      Object {
        "marginBottom": "1rem",
      }
    }
    title="Process"
  >
    <ProcessForm
      graphData={[MockFunction]}
      isGraphValid={true}
      isLoading={true}
      loading={[MockFunction]}
      onSubmitForm={[MockFunction]}
    />
  </Collapsible>
</div>
`;
